packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.js

```diff
@@ -166,10 +167,20 @@ describe('ReactHooksWithNoopRenderer', () => {
     expect(ReactNoop.getChildren()).toEqual([span('Count: 0')]);
 
     // Schedule some updates
-    ReactNoop.batchedUpdates(() => {
-      counter.current.updateCount(1);
-      counter.current.updateCount(count => count + 10);
-    });
+    if (gate(flags => flags.enableSyncDefaultUpdates)) {
+      React.unstable_startTransition(() => {
+        // TODO: Batched updates need to be inside startTransition?
```

Comment:
What's going on here? Seems like a bug

Reply:
We're not taking the `executionContext` into account when scheduling the sync updates, so the updates inside of batchedUpdates are each flushed synchronously

This would fix it in `ensureRootIsScheduled`, but I don't know enough about batchedUpdates or executionContext to know if this is the desired behavior/fix:

<img width="718" alt="Screen Shot 2021-04-09 at 7 10 58 PM" src="https://user-images.githubusercontent.com/2440089/114249118-abec3f80-9967-11eb-83f5-2258e2e5478c.png">

Reply:
Let's follow up with this, I want to merge so that I don't get another merge conflict since this is so big.

Reply:
Nvm, I think this is fine. `batchedUpdates` doesn't change the priority, so it makes sense you need to wrap these with startTransition when the sync-by-default flag is on.

I had misinterpreted the TODO as saying that without `startTransition`, the updates wouldn't be batched.